replica/table.cc

```diff
@@ -826,20 +826,24 @@ table::do_add_sstable_and_update_cache(sstables::shared_sstable sst, sstables::o
             add_maintenance_sstable(cg, sst);
         }
         update_stats_for_new_sstable(sst);
+        if (trigger_compaction) {
+            try_trigger_compaction(cg);
+        }
     }), dht::partition_range::make({sst->get_first_decorated_key(), true}, {sst->get_last_decorated_key(), true}));
 }
 
 future<>
 table::add_sstable_and_update_cache(sstables::shared_sstable sst, sstables::offstrategy offstrategy) {
-    co_await do_add_sstable_and_update_cache(std::move(sst), offstrategy);
-    trigger_compaction();
+    bool do_trigger_compaction = offstrategy == sstables::offstrategy::no;
+    co_await do_add_sstable_and_update_cache(std::move(sst), offstrategy, do_trigger_compaction);
```

Comment:
Maybe we should rename this to import_sstable. The normal compaction behavior is handled in compaction_group IIUC.

can offstrategy be no here?

Reply:
When called from `table_populator::populate_subdir` is may be "no" under some conditions
And with https://github.com/scylladb/scylladb/pull/16863 it might also be "no" for repair with TWCS.